server.js

```diff
@@ -36,37 +76,108 @@ server.post('/api/register', (req, res) => {
         });
 });
 
-server.get('/api/users', (req, res) => {
-    db('users_table').select('id', 'username').then(users => {
-        res.status(201).json(users);
-    }).catch(err => {
-        console.log("error:", err);
-        res.status(500).json(err);
-    })
-});
-
-
 server.post('/api/login', (req, res) => {
+    //grab creds
     const creds = req.body;
 
+    //find the user
     db('users_table')
     .where({username: creds.username})
     .first()
     .then(user => {
+        //check creds
         if (user && bcrypt.compareSync(creds.password, user.password)){
-            res.status(200).send('Welcome');
+            req.session.username = user.username;
+            res.status(200).send(`Welcome ${req.session.username}`);
         } else {
             res.status(401).json({message: 'incorrect combination'});
         }
-    }).catch(err => {res.status(500).send(err, "Everything failed")});
-})
+    }).catch(err => {
+        console.log('/api/login Post error:', err);
+        res.status(500).send(err, "Everything failed")});
+});
+
+server.get('/api/logout', (req, res) => {
+    if (req.session) {
+        req.session.destroy(err => {
+            if (err) {
+                res.send('error logging out');
+            } else {
+                res.send('good bye');
+            }
+        });
+    }
+});
+
+server.get('/api/users', restricted, (req, res) => {
+    
+    if(req.session && req.session.username){
+
```

Comment:
Notice in line 112 you have the `restricted` middleware listed. Do you see any issue with this and line 114? Let me know!